hcn/hcnsupport.go

```diff
@@ -43,17 +50,40 @@ type ApiSupport struct {
 	V2 bool `json:"V2"`
 }
 
+// GetCachedSupportedFeatures returns the features supported by the Service and an error if the query failed. If this has been called
+// before it will return the supported features and error received from the first call. This can be used to optimize if many calls to the
+// various hcn.IsXSupported methods need to be made.
+func GetCachedSupportedFeatures() (SupportedFeatures, error) {
+	// Only query the HCN version and features supported once, instead of everytime this is invoked. The logs are useful to
+	// debug incidents where there's confusion on if a feature is supported on the host machine. The sync.Once helps to avoid redundant
+	// spam of these anytime a check needs to be made for if an HCN feature is supported. This is a common occurrence in kube-proxy
+	// for example.
+	featuresOnce.Do(func() {
+		supportedFeatures, featuresErr = getSupportedFeatures()
+	})
+
+	return supportedFeatures, featuresErr
+}
+
+// Deprecated: Use GetCachedSupportedFeatures instead.
 // GetSupportedFeatures returns the features supported by the Service.
 func GetSupportedFeatures() SupportedFeatures {
-	var features SupportedFeatures
-
-	globals, err := GetGlobals()
+	features, err := GetCachedSupportedFeatures()
 	if err != nil {
 		// Expected on pre-1803 builds, all features will be false/unsupported
-		logrus.Debugf("Unable to obtain globals: %s", err)
+		logrus.Errorf("Unable to obtain globals: %s", err)
```

Comment:
This error message needs to be updated, and we should log the error using a field.

Reply:
Would WithError be more preferable?

Reply:
Yes, `WithError` is good when we only have a single field which is an error.

Reply:
Done